truckersmp_cli/args.py

```diff
@@ -89,6 +89,10 @@ def check_args_errors():
             logging.info("Unable to find logged in steam user automatically.")
             sys.exit("Need the steam account name (-n name) to update.")
 
+    # Always activate the Windows Steam check when not using Proton
+    if not Args.proton:
+        Args.check_windows_steam = True
+
```

Comment:
Good. I would move it somewhere to reduce branches:

```py
    # make sure proton and wine are using the same default
    if Args.wine:
        if (Args.prefixdir == Dir.default_prefixdir["ats"]
                or Args.prefixdir == Dir.default_prefixdir["ets2"]):
            logging.debug("""Prefix directory is the default while using Wine,
making sure it's the same directory as Proton""")
            Args.prefixdir = os.path.join(Args.prefixdir, "pfx")

        # always activate the Windows Steam check when not using Proton
        Args.check_windows_steam = True
```

Reply:
Hm, I've sorted at "# checks for updating" because the check only takes place at that branch.

Reply:
Also we can't check for `Args.wine`. We have to check for `not Args.proton` because we're allowing `--wine` and `--proton` while solely updating.

While looking at it I'm not sure what steamcmd is downloading when using wine. We haven't specified a `+@sSteamCmdForcePlatformType`.

I guess we should deny `--wine` and `--proton` even for updating?

Reply:
> Also we can't check for `Args.wine`. We have to check for `not Args.proton` because we're allowing `--wine` and `--proton` while solely updating.

> I guess we should deny `--wine` and `--proton` even for updating?

The current `update_game()` assumes that *either* `--wine` or `--proton` is specified.

I guess that almost all Proton users would not want to update games for Wine at the same time.

> While looking at it I'm not sure what steamcmd is downloading when using wine. We haven't specified a `+@sSteamCmdForcePlatformType`.

`+@sSteamCmdForcePlatformType` is always `windows` (It can be omitted when Windows version of SteamCMD is used on Wine, though). See `steamcmd_args` in `update_game()`.

Reply:
> > While looking at it I'm not sure what steamcmd is downloading when using wine. We haven't specified a `+@sSteamCmdForcePlatformType`.
> 
> `+@sSteamCmdForcePlatformType` is always `windows` (It can be omitted when Windows version of SteamCMD is used on Wine, though). See `steamcmd_args` in `update_game()`.

Yes, but Proton doesn't exist for windows. Anyway, it doesn't make sense in the first place to use wine to download proton hence why I changed the exclusivity check for `--wine` and `--proton`.

Reply:
> ... to use wine to download proton ...

Proton itself is installed via `steamcmd.sh` only when `--proton` is specified, without specifying `+@sSteamCmdForcePlatformType windows`.

This is based on your "Proton wrapper" (PR 24) and you're the original author :)

Reply:
Too many changes, hard to remember everything ;)